plugin/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -664,4 +667,47 @@ public boolean isAllowed(String groovyResourceUrl) {
         assertThat(logger.getMessages(), containsInRelativeOrder("finished suspending all executions", "ensuring all executions are saved"));
     }
 
+    @Issue("JENKINS-71692")
+    @Test public void stepsAreStoppedWhenCpsVmCroaks() throws Throwable {
+        sessions.then(r -> {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                "parallel(\n" +
+                "  'willCroak': {\n" +
+                "    node {\n" +
+                "      label: {\n" + // Defines a label on a BlockStatement, which is erroneously ignored by CpsTransformer.
+                "        while (true) {\n" +
+                "          semaphore('croak')\n" +
+                "          break label\n" + // CallEnv.getBreakAddress throws an IllegalStateException when this runs.
+                "        }\n" +
+                "      }\n" +
+                "    }\n" +
+                "  },\n" +
+                "  'keepsRunning': {\n" +
+                "    semaphore('other')\n" +
+                "    echo 'kept running!'\n" +
+                "  }\n" +
```

Comment:
Not necessary to reproduce the main problem with `node`, but this shows how steps could still "run" even after `CpsFlowExecution.croak`.